File: python/1_multiple_3_5.py
#!/usr/bin/env python
import sys

def timereps(reps, func):
    from time import time
    start = time()
    for i in range(0, reps):
        func()
    end = time()
    return (end - start) / reps

def print_result():
   result = original(1000)
   time = timereps(10, lambda: original(1000))
   print("Result: {0}, in {1}".format(result, time))

def original(limit):
	return sum([x for x in range(limit) if x % 3 == 0 or x % 5 == 0])

if __name__ == '__main__':
	sys.exit(print_result())

File: erlang/1_multiple_3_5.erl
#!/usr/bin/env escript

main([String]) ->
  try
  	N = list_to_integer(String),
  	statistics(runtime),
  	Result = original(N),
  	{_, Time} = statistics(runtime),
  	io:format("Result: ~w in ~p microsecs~n", [Result, Time])
  catch
  	_:_ ->
  	  usage()
  end;

main(_) ->
  usage().

usage() ->
  io:format("usage: 1_multiple_3_5 1000"),
  halt(1).

% Functional code
original(Number) ->
  L = [X || X <- lists:seq(1, Number - 1), (X rem 3 =:= 0) or (X rem 5 =:= 0)],
  lists:sum(L).
